lib/semantics/type.cc

```diff
@@ -269,34 +269,22 @@ std::ostream &operator<<(std::ostream &o, const DeclTypeSpec &x) {
   }
 }
 
-ProcInterface::ProcInterface(const ProcInterface &that)
-  : symbol_{that.symbol_} {
-  if (that.type_) {
-    set_type(*that.type_);
-  }
-}
-ProcInterface &ProcInterface::operator=(ProcInterface &&that) {
-  symbol_ = that.symbol_;
-  type_ = std::move(that.type_);
-  return *this;
-}
-
 void ProcInterface::set_symbol(const Symbol &symbol) {
   CHECK(!type_);
   symbol_ = &symbol;
 }
 void ProcInterface::set_type(const DeclTypeSpec &type) {
   CHECK(!symbol_);
-  type_ = std::make_unique<DeclTypeSpec>(type);
+  type_ = type;
```

Comment:
If symbol_ and type_ are mutually exclusive, consider using a std::variant.

Reply:
This is still evolving, but that's probably a good idea.